File: Lab5/C#/lab5.2/Letters.cs
using System;


namespace lab5._2
{
    class Letters: Str
    {
        public Letters() : base() { }
        public Letters(string s) : base(s) { }


        public override void Lenght()
        {
            int lettersInStr = 0;
            for(int i = 0; i < str.Length; i++)
            {
                if (Char.IsLetter(str[i]))
                {
                    lettersInStr++;
                }
                else { error = true; }
            }
            Len = lettersInStr;
        }

        public override void Processing()
        {
            char[] charArray = str.ToCharArray();
            Array.Reverse(charArray);
            this.str = new string(charArray);
            this.Lenght();
        }
    }
}

File: Lab5/C++/lab5.2/Letters.cpp
#include "Letters.h"

Letters::Letters() : Str() { };
Letters::Letters(string s) { str = s; LenCount(s); };

void Letters::LenCount(string s)
{
	int tempLen = 0;
	error = false;
	for (int i = 0; i < s.length(); i++)
	{
		if (isalpha(s[i])) {
			tempLen++;
		}
		else error = true;
	}
	len = tempLen;
}
void Letters::Processing()
{
	reverse(str.begin(), str.end());
}